File: app/src/main/java/com/proxerme/app/receiver/NotificationReceiver.java
package com.proxerme.app.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.proxerme.app.service.NotificationService;

/**
 * Receiver for the {@link NotificationService}.
 *
 * @author Ruben Gees
 */
public class NotificationReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        //noinspection WrongConstant
        NotificationService.load(context, intent.getAction());
    }
}

File: app/src/main/kotlin/com/proxerme/app/receiver/NotificationReceiver.kt
package com.proxerme.app.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.proxerme.app.service.NotificationService

/**
 * Receiver for the [NotificationService].

 * @author Ruben Gees
 */
class NotificationReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        NotificationService.load(context, intent.action)
    }
}
